go/src/koding/klient/app/klient.go

```diff
@@ -461,11 +465,11 @@ func (k *Klient) RegisterMethods() {
 	k.kite.HandleFunc("klient.usage", k.usage.Current)
 
 	// klient os method(s)
-	k.kite.HandleFunc("os.home", kos.Home)
-	k.kite.HandleFunc("os.currentUsername", kos.CurrentUsername)
+	k.handleWithSub("os.home", kos.Home)
```

Comment:
How about making this more configurable?

Create a configuration map [like this one](https://github.com/koding/koding/pull/9882/commits/1389cf689dcee7f2cd0d1fed465fbe33a7e82bb1#diff-514e03b9d69e198098d3560e3bda3763R213), and register a PreHandlerFunc that if sub-check is requested, will validate it and return error if sub is not valid. Basically would work the same as auth check etc.